scripts/build-unix.sh

```diff
@@ -30,8 +30,11 @@ fi
 # enable all compiler warnings
 CXXFLAGS="$CXXFLAGS -Wall"
 
-if [ "$WARN_AS_ERR" = "ON" ]; then
+if [ "${BUILD_TYPE,,}" = "debug" ]; then
```

Comment:
```suggestion
if [ "${BUILD_TYPE}" = "debug" ]; then
```
why do the commas exist?

Reply:
It lower-cases the variable as a way of doing a case-insensitive string comparison.

https://stackoverflow.com/a/19411918

Reply:
> I tried on my macbook bash shell and it failed:

It's a bash 4.0 feature apparently... which is a problem for the MacOS runner.

> Do we need case insensitive comparison at all?

No, I just thought it seemed safer - removed - 6d44615698ec71cc4b47402bfcd3f9e5c1c5c68a